test/e2e/plugin_lifecycle/plugin_lifecycle_test.go

```diff
@@ -618,4 +618,45 @@ var _ = framework.CLICoreDescribe("[Tests:E2E][Feature:Plugin-lifecycle]", func(
 			}
 		})
 	})
+
+	// use case: Plugin inventory DB cache refresh check on every tanzu command
+	// and perform refresh only when digest timestamp is passed 24 hours or env variable TANZU_CLI_DATABASE_REFRESH_THRESHOLD is set
+	// a. run a "tanzu plugin source init" to do a digest check
+	// b. set the TTL to 5 seconds
+	// c. set the TANZU_CLI_DATABASE_REFRESH_THRESHOLD to 10 seconds
+	// d. Run a tanzu plugin list every 2 seconds and looking for the DB refresh printout after the 5th command.
+
+	Context("plugin inventory DB cache refresh when digest timestamp is passed the expiry threshold", func() {
+		const (
+			refreshingDBPrintout           = "Reading plugin inventory for"
+			refreshingPluginInventoryCache = "Refreshing plugin inventory cache"
+		)
+
+		It("plugin list doesnt refresh db cache until threshold is passed", func() {
+
+			// Set the TTL to something small: 5 seconds
+			_ = os.Setenv(constants.ConfigVariablePluginDBCacheTTL, "5")
+			_ = os.Setenv(constants.ConfigVariablePluginDBCacheRefreshThreshold, "5s")
+
+			for i := 0; i < 5; i++ {
```

Comment:
I suggest looping 8 times, so the TTL is expired and that we have seen that at second 6 and 7 and 8, there was no refresh due to the threshold (assuming a 10 sec threshold set above)

Reply:
Thanks for the suggestion, updated the test accordingly.